### PR TITLE
CustomProDB update for PSM2SAM and MVP

### PR DESCRIPTION
**Pull request by chambm**
* Update customProDB to use my GitHub codebase (currently 1.16.0)
  * requires adding many conda requirements; if official customProDB merges my changes, these can be removed
* Make output_indel.fasta optional
* Add RData variant annotation output for PSM2SAM
* Add SQLite genomic mapping and variant annotation outputs for Galaxy-P MVP viewer
* Add tests using dbSNP and COSMIC annotations and reorganize test files to make it clear they are only for hg19
  * TODO: add tests for different genomes

### CODE REVIEW

File: tools/bumbershoot/custom_pro_db/tool-data/customProDB.loc.sample [hunk 1]
```diff
@@ -1,18 +0,0 @@
-#This file lists the locations and dbkeys of all the fasta files
-#under the "genome" directory (a directory that contains a directory
```
Comment (bgruening) [inline tools/bumbershoot/custom_pro_db/tool-data/customProDB.loc.sample]: You are still using the data table, isn't it?

**Comment by jj-umn** [comment]
@chambm @bgruening 
Here is the schema for genomic mapping, for the CDS sequence of the variant Search DB protein
( with name being the the one supplied in the Search DB ):
    sqlite> .schema
    CREATE TABLE feature_cds_map (
        name TEXT,
        cds_start INTEGER,
        cds_end INTEGER,
        chrom TEXT,
        start INTEGER,
        end INTEGER,
        strand TEXT
    );

This is similar to GFF in that contiguous genome-mapped region is a single row in the table.  Genomic locations for amino acids in the protein can be queried.   In addition, proteins in a genomic region can be queried.  

This schema allows us to combine customProDB generated variants with structural variants from RNAseq structural variant applications such as deFuse and StarFusion.    Section 5.4 of https://github.com/samtools/hts-specs/blob/master/VCFv4.3.pdf
shows the variety of variants we'd like to be able to contain in the genomic mapping table.  I've worked with a few cancer researchers over past few years and providing a convenient jump from a protein-centric view to these more complicated genomics locations can be very useful.  

The reason I put in the "CDS" start and end is to allow for cases where only parts of the constructed proteins search sequence can be mapped to a reference. 


Assume:

#CHROM  POS     ID     REF  ALT               QUAL  FILTER  INFO
2       321682  bnd_V  T    ]13:123456]AGTCAT 6     PASS    SVTYPE=BND;MATEID=bnd_U
13      123456  bnd_U  C    C'[2:321682[  6     PASS    SVTYPE=BND;MATEID=bnd_V

We take a portion from chr13 before the break:  chr13: chr13:123430-123456  GAGACAGCCGAGAGTAGGGAGGTGGC
We have the inserted sequence (exluding the REF matching parts):  CAGTCA
We take a portion from chr2 after the break: chr2:321682-321704  TTCCACATGGCTGGGGAGGTTT
The constructed CDS: GAGACAGCCGAGAGTAGGGAGGTGGCCAGTCATTCCACATGGCTGGGGAGGTTT
The Search DB Protein: 
>Variant_1
ETAESREVASHSTWLGRF

The genomic mapping:
Variant_1    0    26    chr13    123430    123456    +
Variant_1    32   54    chr2     321682    321704    +   

When MVP displays the genomic mapping bars over the Variant_1 AA sequence
 there will be a gap over AAs: AS
since they are an insertion with no known genomic mapping

File: tools/bumbershoot/custom_pro_db/tool-data/customProDB.loc.sample (hunk 1 above)
Comment (chambm) [inline tools/bumbershoot/custom_pro_db/tool-data/customProDB.loc.sample]: I do use the data table, but it's created by the data_manager. Planemo broke when I deleted `tool_data_table_conf.xml.sample`, but it still works with the tool-data directory deleted. Is there another reason to keep it?

Comment (chambm) [inline tools/bumbershoot/custom_pro_db/tool-data/customProDB.loc.sample]: Planemo/Travis did not detect this error condition properly:
```
2017-05-04 20:22:44,984 WARNI [galaxy.tools.parameters.dynamic_options] Data table named 'customProDB' is required by tool but not configured
2017-05-04 20:22:44,984 ERROR [galaxy.tools.toolbox.base] Failed to load potential tool /home/travis/build/galaxyproteomics/tools-galaxyp/tools/bumbershoot/custom_pro_db/customProDB.xml.
Traceback (most recent call last):
  File "/tmp/tmpW9tsUo/galaxy-dev/lib/galaxy/tools/toolbox/base.py", line 701, in quick_load
    tool = self.load_tool( tool_file )
  File "/tmp/tmpW9tsUo/galaxy-dev/lib/galaxy/tools/toolbox/base.py", line 736, in load_tool
    tool = self.create_tool( config_file=config_file, repository_id=repository_id, guid=guid, **kwds )
  File "/tmp/tmpW9tsUo/galaxy-dev/lib/galaxy/tools/__init__.py", line 249, in create_tool
    tool = ToolClass( config_file, tool_source, self.app, guid=guid, repository_id=repository_id, **kwds )
  File "/tmp/tmpW9tsUo/galaxy-dev/lib/galaxy/tools/__init__.py", line 425, in __init__
    raise e
KeyError: 'customProDB'
2017-05-04 20:22:44,986 INFO  [galaxy.tools.toolbox.base] Parsing the tool configuration /tmp/tmpW9tsUo/shed_tools_conf.xml
2017-05-04 20:22:44,987 INFO  [galaxy.tools.toolbox.base] Parsing the tool configuration /tmp/tmpW9tsUo/empty_tool_conf.xml
No handlers could be found for logger "galaxy.datatypes.display_applications.application"
2017-05-04 20:22:45,487 WARNI [galaxy.webhooks] directory not found: /tmp/tmpW9tsUo/galaxy-dev/lib/galaxy/util/../../.././config/plugins/webhooks/demo/config
2017-05-04 20:22:45,502 INFO  [galaxy.jobs] Handler 'main' will load all configured runner plugins
2017-05-04 20:22:45,512 INFO  [galaxy.jobs.handler] job handler stop queue started
2017-05-04 20:22:45,529 INFO  [galaxy.jobs.handler] job handler queue started
2017-05-04 20:22:46,650 INFO  [galaxy.queue_worker] Binding and starting galaxy control worker for main
/home/travis/.planemo/gx_venv/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:562: SAWarning: Dialect sqlite+pysqlite does *not* support Decimal objects natively, and SQLAlchemy must convert from floating point - rounding errors and other issues may occur. Please consider storing Decimal numbers as strings or integers on this platform for lossless storage.
  'storage.' % (dialect.name, dialect.driver))
/home/travis/.planemo/gx_venv/lib/python2.7/site-packages/sqlalchemy/sql/default_comparator.py:153: SAWarning: The IN-predicate on "request_type.id" was invoked with an empty sequence. This results in a contradiction, which nonetheless can be expensive to evaluate.  Consider alternative strategies for improved performance.
  'strategies for improved performance.' % expr)
----------------------------------------------------------------------
XML: /tmp/tmpW9tsUo/xunit.xml
----------------------------------------------------------------------
Ran 0 tests in 0.001s
```

Comment (bgruening) [inline tools/bumbershoot/custom_pro_db/tool-data/customProDB.loc.sample]: This is needed afaik just to make this toolshed repo aware of the data_table. You can not upload this tool to the toolshed without these files/structure.

**Comment by chambm** [comment]
Currently waiting on conda-forge/staged-recipes/pull/2952

**Comment by jj-umn** [comment]
@chambm @bgruening I see conda-forge/staged-recipes#2952 got merged.  Is there any way to rerun the test without doing another commit?

**Comment by bgruening** [comment]
You can restart every travis job on the travis website, or simple close and reopen a PR.

**Comment by chambm** [comment]
The package isn't uploaded to conda yet. Also, it may get built for the wrong R version. :-/

**Comment by chambm** [comment]
I love it when a plan comes together. Even if it takes a while. :laughing:

**Comment by bgruening** [comment]
Great work @chambm! Sorry for the hassle we are working to improve it.

**Pull request merged**